### PR TITLE
Major Rainloop update

### PR DESCRIPTION
**Pull request by scith**
- Backup/restore scripts
- Updated to YNH 2.4
- Fetch from source and integrity check
- Remove autoversion (not needed)
- Remove ynh_login_mapping (creates LDAP errors plus is it really
needed?). The plugin is still installed but not activated
- Domain hooks
- Backup before upgrade, restore if fail
- Remove disabled domains (users may want to add gmail addresses)
- Added auto-domain-grab plugin with wildcard domain. Users can try to
add aliases without admin config
- Add the admin password in config in case the password is lost
- Adds a /pgpback path to backup/restore PGP private keys on the server from browser storage using [PGPback by chtixof](https://github.com/chtixof/pgpback_ynh)

### CODE REVIEW

**Comment by scith** [comment]
What do you think? I need some testers
``sudo yunohost app upgrade rainloop -u https://github.com/scith/rainloop_ynh --verbose``

**Review by M5oul — APPROVED** [review]
Upgrade went fine.

File: scripts/upgrade [hunk 1]
```diff
@@ -1,48 +1,76 @@
...
+	sudo yunohost backup delete $app-before-upgrade
+	sudo yunohost backup create --ignore-hooks --apps $app --name $app-before-upgrade --quiet
+	EXIT_PROPERLY () {
+		trap '' EXIT
+		set +eu
+		sudo yunohost backup restore --ignore-hooks $app-before-upgrade --apps $app --force --quiet	# Restore the backup if upgrade failed
+		ynh_die "Upgrade failed. The app was restored to the way it was before the failed upgrade."
+	}
+	set -eu	
+	trap EXIT_PROPERLY ERR
```
Comment (maniackcrudelis) [inline scripts/upgrade:15]: Ici ça ne fonctionnera pas.
trap capture les signaux d'erreur, mais set -u quittera en cas d'erreur.
Donc trap sera doublé par set -u.

Pour pallier ça, tu dois utiliser un trap EXIT pour capturer la sortie du script. Mais dans ce cas, EXIT_PROPERLY se déclenchera à la sortie du script, que ce soit une erreur ou pas.
Il faut donc aussi vérifier le code de sortie, comme ici:
https://github.com/maniackcrudelis/modele_ynh/blob/master/scripts/.fonctions#L10-L13

Comment (maniackcrudelis) [reply scripts/upgrade:15]: Et il faut un remove de l'app avant le restore, sinon il échoue (et silencieusement à cause du --quiet)

Comment (scith) [reply scripts/upgrade:15]: Merci, je vais explorer ça

**Comment by polytan02** [comment]
The php-fpm was not backed up so it failed the restore. I added it.

If you are OK with the commit, please can you update the community.json so that everybody can upgrade their app ?

**Pull request merged**